defaults/main.yml

```diff
@@ -32,8 +32,10 @@ amq_broker_skip_configure: false
 amq_broker_write_amq_facts: false
 
 # Path to locally generated facts about amq broker [/etc/ansible/facts.d/]
-amq_broker_default_facts_file: /etc/ansible/facts.d/amq_broker.fact
-
+#amq_broker_default_facts_file: /etc/ansible/facts.d/amq_broker.fact
+amq_broker_default_facts_file: "{{ lookup('env','AMQ_BROKER_FACTS_NODE_DIR') | default('/tmp/facts/', true) }}/amq_broker_facts_{{ external_address | default(ansible_default_ipv4.address) }}"
```

Comment:
I think we should default to /var instead of /tmp. Data on tmp is transient. It is removed on reboot ... On var is it not. There may be situations where we might need to reboot the broker host, but we still want to keep the facts file between reboots.

Reply:
Here I would like to differ. The facts are not only "static ones", but are dependent on current profile/deployment.
For example, facts will contain also HA/Clustered status (master/slave, extra params), shared store, etc.

So these are not only ansible_facts, but mostly facts related to dtests ExteranlBroker, to know, what configuration is deployed, so we can operate it properly.

Please see templates/amq_broker_facts_template.j2 for details